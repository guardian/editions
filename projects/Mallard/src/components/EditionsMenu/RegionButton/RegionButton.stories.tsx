import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { withKnobs } from '@storybook/addon-knobs'
import { RegionButton } from './RegionButton'

storiesOf('RegionButton', module)
    .addDecorator(withKnobs)
    .add('RegionButton - default', () => (
        <RegionButton
            onPress={() => {}}
            title="Daily Edition"
            subTitle={'Published every day by 6am (GMT)'}
        />
    ))
    .add('RegionButton - selected', () => (
        <RegionButton
            selected
            onPress={() => {}}
            title={'Daily Edition'}
            subTitle={'Published every day by 6am (GMT)'}
        />
    ))
