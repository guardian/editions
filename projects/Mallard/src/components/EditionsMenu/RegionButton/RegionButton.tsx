import React, { useState } from 'react'
import { StyleSheet, Text, TouchableWithoutFeedback, View } from 'react-native'
import { TitlepieceText } from 'src/components/styled-text'
import { color } from 'src/theme/color'
import { families } from 'src/theme/typography'
import { EDITIONS_MENU_TEXT_LEFT_PADDING } from '../EditionsMenu'

const styles = (selected: boolean) =>
    StyleSheet.create({
        container: {
            backgroundColor: selected
                ? color.primary
                : color.palette.neutral[97],
            paddingBottom: 32,
            paddingLeft: EDITIONS_MENU_TEXT_LEFT_PADDING,
            paddingTop: 4,
        },
        title: {
            fontSize: 20,
            lineHeight: 24,
            marginBottom: 5,
            color: selected ? 'white' : color.primary,
        },
        subTitle: {
            fontFamily: families.sans.regular,
            fontSize: 15,
            lineHeight: 18,
            color: selected ? 'white' : color.text,
        },
    })

const RegionButton = ({
    onPress,
    selected = false,
    subTitle,
    title,
}: {
    onPress: () => void
    selected?: boolean
    subTitle: string
    title: string
}) => {
    const [pressed, setPressed] = useState(false)
    const defaultStyles = styles(selected || pressed)

    return (
        <TouchableWithoutFeedback
            accessibilityRole="button"
            accessibilityLabel={`${title} - Regional Edition Button`}
            onPressIn={() => setPressed(true)}
            onPressOut={() => setPressed(false)}
            onPress={onPress}
        >
            <View style={defaultStyles.container}>
                <TitlepieceText style={defaultStyles.title}>
                    {title}
                </TitlepieceText>
                <Text style={defaultStyles.subTitle}>{subTitle}</Text>
            </View>
        </TouchableWithoutFeedback>
    )
}

export { RegionButton }
