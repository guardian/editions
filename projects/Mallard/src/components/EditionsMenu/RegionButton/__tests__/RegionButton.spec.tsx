import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { RegionButton } from '../RegionButton'

const props = {
    onPress: () => {},
    title: 'Australia Weekend',
    subTitle: 'Published every Saturday by 6am (AEST)',
}

describe('RegionButton', () => {
    it('should display a default RegionButton with correct styling', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <RegionButton {...props} />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should display a selected RegionButton with correct styling', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <RegionButton selected={true} {...props} />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
