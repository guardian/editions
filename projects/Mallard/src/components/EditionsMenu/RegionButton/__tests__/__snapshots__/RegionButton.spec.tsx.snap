// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionButton should display a default RegionButton with correct styling 1`] = `
<View
  accessibilityLabel="Australia Weekend - Regional Edition Button"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "paddingBottom": 32,
      "paddingLeft": 104,
      "paddingTop": 4,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 30,
          "lineHeight": 33,
        },
        Object {
          "color": "#052962",
          "fontSize": 20,
          "lineHeight": 24,
          "marginBottom": 5,
        },
      ]
    }
  >
    Australia Weekend
  </Text>
  <Text
    style={
      Object {
        "color": "#121212",
        "fontFamily": "GuardianTextSans-Regular",
        "fontSize": 15,
        "lineHeight": 16,
      }
    }
  >
    Published every Saturday by 6am (AEST)
  </Text>
</View>
`;

exports[`RegionButton should display a selected RegionButton with correct styling 1`] = `
<View
  accessibilityLabel="Australia Weekend - Regional Edition Button"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#052962",
      "paddingBottom": 32,
      "paddingLeft": 104,
      "paddingTop": 4,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 30,
          "lineHeight": 33,
        },
        Object {
          "color": "white",
          "fontSize": 20,
          "lineHeight": 24,
          "marginBottom": 5,
        },
      ]
    }
  >
    Australia Weekend
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "GuardianTextSans-Regular",
        "fontSize": 15,
        "lineHeight": 16,
      }
    }
  >
    Published every Saturday by 6am (AEST)
  </Text>
</View>
`;
